File: 2025/253. Meeting Rooms II/Solution.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        Arrays.sort(intervals, (x, y) -> (x[0] - y[0]));
        Queue<int[]> heap = new PriorityQueue<>((x, y) -> (x[1] - y[1]));
        int max = 0, current = 0;
        for(int[] in : intervals) {
            if (!heap.isEmpty()) {
                if (heap.peek()[1] <= in[0]) {
                    heap.remove();
                }
            }  
            heap.offer(in);
            max = Math.max(max, heap.size());
        }
        return max;
    }
}

File: 2025/253. Meeting Rooms II/solution.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        min_heap = []
        res = 0
        for start, end in intervals:
            if min_heap and min_heap[0] <= start:
                heappop(min_heap)
            heappush(min_heap, end)
        return len(min_heap)